### PR TITLE
Created LocationType & Include it into Location

### LINKED ISSUE #34: Development: New Field Location Type on the Location API
Now that the location API has been finished, include the new Location Type fields in it.

### PR DESCRIPTION
**Pull request by Charhmwang**
LocationType route has been created and included into Location.
There are two urls working as expected:
localhost:8080/ds/v1/location_type/all
localhost:8080/ds/v1/location_type/by_id/COMPOUND

### CODE REVIEW

File: src/main/java/com/jlr/ttl/ds/api/controllers/LocationTypeController.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
+package com.jlr.ttl.ds.api.controllers;
```
Comment (BFARIAAL) [inline src/main/java/com/jlr/ttl/ds/api/controllers/LocationTypeController.java]: The ticket was to include location type in the location api, not to create a new one. Please delete this file

File: src/main/java/com/jlr/ttl/ds/api/dto/entity/LocationType.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package com.jlr.ttl.ds.api.dto.entity;
```
Comment (BFARIAAL) [inline src/main/java/com/jlr/ttl/ds/api/dto/entity/LocationType.java]: Delete this file

File: src/main/java/com/jlr/ttl/ds/api/dto/request/LocationTypeRequest.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package com.jlr.ttl.ds.api.dto.request;
```
Comment (BFARIAAL) [inline src/main/java/com/jlr/ttl/ds/api/dto/request/LocationTypeRequest.java]: Delete this file

File: src/main/java/com/jlr/ttl/ds/api/dto/response/LocationTypeResponse.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package com.jlr.ttl.ds.api.dto.response;
```
Comment (BFARIAAL) [inline src/main/java/com/jlr/ttl/ds/api/dto/response/LocationTypeResponse.java]: Delete this file

File: src/main/java/com/jlr/ttl/ds/api/dto/table/LocationTypeTable.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
+package com.jlr.ttl.ds.api.dto.table;
```
Comment (BFARIAAL) [inline src/main/java/com/jlr/ttl/ds/api/dto/table/LocationTypeTable.java]: Delete this file

File: src/main/java/com/jlr/ttl/ds/api/exception/data/LocationTypeNotFoundException.java [hunk 6]
```diff
@@ -0,0 +1,5 @@
+package com.jlr.ttl.ds.api.exception.data;
```
Comment (BFARIAAL) [inline src/main/java/com/jlr/ttl/ds/api/exception/data/LocationTypeNotFoundException.java]: Delete this file

File: src/main/java/com/jlr/ttl/ds/api/repositories/LocationTypeRepository.java [hunk 7]
```diff
@@ -0,0 +1,11 @@
+package com.jlr.ttl.ds.api.repositories;
```
Comment (BFARIAAL) [inline src/main/java/com/jlr/ttl/ds/api/repositories/LocationTypeRepository.java]: Delete this file

File: src/main/java/com/jlr/ttl/ds/api/services/LocationTypeService.java [hunk 8]
```diff
@@ -0,0 +1,73 @@
+package com.jlr.ttl.ds.api.services;
```
Comment (BFARIAAL) [inline src/main/java/com/jlr/ttl/ds/api/services/LocationTypeService.java]: Delete this file

File: src/main/java/com/jlr/ttl/ds/api/util/mapper/LocationTypeValueMapper.java [hunk 9]
```diff
@@ -0,0 +1,16 @@
+package com.jlr.ttl.ds.api.util.mapper;
```
Comment (BFARIAAL) [inline src/main/java/com/jlr/ttl/ds/api/util/mapper/LocationTypeValueMapper.java]: Delete this file

**Review by BFARIAAL — CHANGES_REQUESTED** [review]
This PR has a lot of changes that are out of the scope of its ticket. The ticket was to include location type in the location API and not to create a brand new API. Please resolve this issue

File: src/main/java/com/jlr/ttl/ds/api/util/mapper/LocationTypeValueMapper.java (hunk 9 above)
Comment (Charhmwang) [reply src/main/java/com/jlr/ttl/ds/api/util/mapper/LocationTypeValueMapper.java]: All the files are out of this ticket's scope have been removed.

**Review by BFARIAAL — APPROVED** [review]

**Pull request merged**